internal/db/db.go

```diff
@@ -39,7 +39,8 @@ import (
 )
 
 var (
-	log = corelog.NewLogger("db")
+	log    = corelog.NewLogger("db")
+	tracer = otel.Tracer("github.com/sourcenetwork/defradb/internal/db")
```

Comment:
thought: I think I agree with John's suggestion of extracting this out to a thin package so we can avoid scattering references throughout the code base to the same 3rd party package.  

If we ever swap it out we'll probably still need to change the func signatures (I wouldn't put much thought into designing them), but at least everything will be in the same place, and it makes adding/standardising any middleware much easier.

Reply:
Update: the metrics were all moved to the internal `metric` package and put behind a build flag `telemetry`.